AsyncTcpClient/AsyncTcpClient.cs

```diff
@@ -234,11 +234,16 @@ public async Task RunAsync()
 							Message?.Invoke(this, new AsyncTcpEventArgs("Connection reset remotely", ex));
 							readLength = -2;
 						}
+						catch(Exception ex)
+						{
+							Message?.Invoke(this, new AsyncTcpEventArgs("Read from stream failed", ex));
+							readLength = -3;
+						}
 						if (readLength <= 0)
 						{
 							if (readLength == 0)
 							{
-								Message?.Invoke(this, new AsyncTcpEventArgs("Connection closed remotely"));

```

Comment:
This message will be invoked even if connection was disconnected locally because readAsync always return 0